extensions/amp-digidip/0.1/amp-digidip.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {LinkShifter} from './link-shifter';
+import {Services} from '../../../src/services';
+import {getConfigOpts} from './config-options';
+import {getScopeElements} from './helper';
+
+export class AmpDigidip extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampDoc_ = null;
+
+    /** @private {?../../../src/service/viewer-impl.Viewer} */
+    this.viewer_ = null;
+
+    /* @private {?./link-shifter} */
+    this.shifter_ = null;
+
+    /** @private {?Object} */
+    this.configOpts_ = {};
+
+    /** @private {?Object} */
+    this.listElements_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+
+    this.ampDoc_ = this.getAmpDoc();
+    this.viewer_ = Services.viewerForDoc(this.ampDoc_);
+
+    this.configOpts_ = getConfigOpts(this.element);
+
+    return this.ampDoc_.whenBodyAvailable()
+        .then(() => {
+          this.letsRockIt_();
+        });
+  }
+
+  /**
+   * @private
+   */
+  letsRockIt_() {
+    this.shifter_ = new LinkShifter(
+        this.element,
+        this.viewer_);
+
+    this.attachClickEvent_();
+  }
+
+  /**
+   * @private
+   */
+  attachClickEvent_() {
+    this.listElements_ = getScopeElements(
+        this.ampDoc_,
+        this.configOpts_);
+
+    this.listElements_.forEach(nodeElement => {
```

Comment:
AMP runtime provides a global service to add click listener:
https://github.com/ampproject/amphtml/blob/f01d749ba83c1d5af7b62d2906b7bbf0561bfb17/extensions/amp-analytics/0.1/linker-manager.js#L115-L118

Please switch to that.